ops/model.py

```diff
@@ -678,6 +683,8 @@ def open_port(
         Use :meth:`set_ports` for a more declarative approach where all of
         the ports that should be open are provided in a single call.
 
+        .. jujuversion:: 3.1
```

Comment:
I think the ports functionality works quite a ways back for machines -- it was just the support on K8s that was added in 3.0.3. Or am I missing something?

Reply:
Hmm, the [Juju docs](https://juju.is/docs/juju/hook-tool#open-port) have a "Requires Juju 3.1+" note, but then the description does talk about how the behaviour was different in 2.9.

Should I check this with the Juju team, or are you confident that the note in the Juju docs should be fixed, and ours should just have a note about K8s charms getting this later and using jujuversion as a guard?

Reply:
[Matrix](https://matrix.to/#/!xzmWHtGpPfVCXKivIh:ubuntu.com/$v_MYS-BDWb0AIUPClA64eWPEBsZ58zIQEzKSCQ1V5tU?via=ubuntu.com&via=matrix.org&via=matrix.debian.social)